test/mavsdk_tests/autopilot_tester.cpp

```diff
@@ -244,6 +258,8 @@ void AutopilotTester::prepare_square_mission(MissionOptions mission_options)
 	_mission->set_return_to_launch_after_mission(mission_options.rtl_at_end);
 
 	REQUIRE(_mission->upload_mission(mission_plan) == Mission::Result::Success);
+	// PX4 needs time to realize that it now has a mission available, so we need to wait a bit here.
```

Comment:
I'm aware of why we need this, but it would be nice if we didn't have to depend on sleeps for synchronization issues. It feels like a band-aid covering a deep wound (sorry for the gory analogy). Is there anything we can do within PX4 mavlink receiver/publisher to address this? last time I checked (long time ago) this module wasn't lockstep friendly?

Reply:
I would blame this on PX4's architecture:

The mavlink module responds that it has received the mission but it hasn't internally updated state everywhere, so commander isn't aware yet.

Waiting for mission state being updated in MAVSDK is just a different bandaid for a PX4 problem.

Reply:
This may have been an intentional design decision, depending on the original intent. I’ll investigate further and evaluate whether a behavior change is viable.